placement/algo/algo.go

```diff
@@ -0,0 +1,450 @@
+package algo
+
+import (
+	"container/heap"
+	"errors"
+	"math"
+
+	"github.com/m3db/m3cluster/placement"
+)
+
+var (
+	errNotEnoughRacks = errors.New("not enough host to take shards, please make sure RF <= number of racks")
+	errHostAbsent     = errors.New("could not remove or replace a host that does not exist")
+)
+
+type placementAlgorithm struct {
+}
+
+func newM3DBPlacementAlgorithm() placement.Algorithm {
+	return placementAlgorithm{}
+}
+
+func newM3DBDeploymentPlanner() placement.DeploymentPlanner {
+	return placementAlgorithm{}
+}
+
+func (a placementAlgorithm) InitPlacement(hosts []placement.Host, ids []int) (placement.Snapshot, error) {
```

Comment:
Usually ``Init`` is used to initialize an object with state.  I think in this case it is building the initial placement? If so I'd call it ``BuildInitialPlacement`` or something similar.